// @pulumi/kubernetes/authorization/v1beta1/index.js
"use strict";
// *** WARNING: this file was generated by pulumigen. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubjectAccessReviewPatch = exports.SubjectAccessReview = exports.SelfSubjectRulesReviewPatch = exports.SelfSubjectRulesReview = exports.SelfSubjectAccessReviewPatch = exports.SelfSubjectAccessReview = exports.LocalSubjectAccessReviewPatch = exports.LocalSubjectAccessReview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
exports.LocalSubjectAccessReview = null;
utilities.lazyLoad(exports, ["LocalSubjectAccessReview"], () => require("./localSubjectAccessReview"));
exports.LocalSubjectAccessReviewPatch = null;
utilities.lazyLoad(exports, ["LocalSubjectAccessReviewPatch"], () => require("./localSubjectAccessReviewPatch"));
exports.SelfSubjectAccessReview = null;
utilities.lazyLoad(exports, ["SelfSubjectAccessReview"], () => require("./selfSubjectAccessReview"));
exports.SelfSubjectAccessReviewPatch = null;
utilities.lazyLoad(exports, ["SelfSubjectAccessReviewPatch"], () => require("./selfSubjectAccessReviewPatch"));
exports.SelfSubjectRulesReview = null;
utilities.lazyLoad(exports, ["SelfSubjectRulesReview"], () => require("./selfSubjectRulesReview"));
exports.SelfSubjectRulesReviewPatch = null;
utilities.lazyLoad(exports, ["SelfSubjectRulesReviewPatch"], () => require("./selfSubjectRulesReviewPatch"));
exports.SubjectAccessReview = null;
utilities.lazyLoad(exports, ["SubjectAccessReview"], () => require("./subjectAccessReview"));
exports.SubjectAccessReviewPatch = null;
utilities.lazyLoad(exports, ["SubjectAccessReviewPatch"], () => require("./subjectAccessReviewPatch"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "kubernetes:authorization.k8s.io/v1beta1:LocalSubjectAccessReview":
                return new exports.LocalSubjectAccessReview(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:LocalSubjectAccessReviewPatch":
                return new exports.LocalSubjectAccessReviewPatch(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:SelfSubjectAccessReview":
                return new exports.SelfSubjectAccessReview(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:SelfSubjectAccessReviewPatch":
                return new exports.SelfSubjectAccessReviewPatch(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:SelfSubjectRulesReview":
                return new exports.SelfSubjectRulesReview(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:SelfSubjectRulesReviewPatch":
                return new exports.SelfSubjectRulesReviewPatch(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:SubjectAccessReview":
                return new exports.SubjectAccessReview(name, undefined, { urn });
            case "kubernetes:authorization.k8s.io/v1beta1:SubjectAccessReviewPatch":
                return new exports.SubjectAccessReviewPatch(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("kubernetes", "authorization.k8s.io/v1beta1", _module);
